mycore-base/src/main/java/org/mycore/crypt/MCRCryptResolver.java

```diff
@@ -67,7 +67,7 @@ public Source resolve(String s, String s1) throws TransformerException {
         String cipherID = parts[2];
         String value = parts[3];
 
-        String returnString = "";
```

Comment:
Suggestion: set default text and remove from catches.
```
final Element root = new Element("value");
root.setText(action.equals("encrypt") ? "" : value);
```

Reply:
the default text differs from the text in exceptions, removed the String variable and set the element's content directly

Reply:
I meant the following:
```
       final Element root = new Element("value");
       root.setText(action.equals("encrypt") ? "" : value);
       try {
            MCRCipher cipher = MCRCipherManager.getCipher(cipherID);
            root.setText( action.equals("encrypt") ? cipher.encrypt(value) : cipher.decrypt(value));
        } catch (MCRCryptKeyFileNotFoundException e) {
            LOGGER.error(e::getMessage, e);
        } catch (MCRCryptKeyNoPermissionException e) {
            LOGGER.info(() -> "No permission to read cryptkey" + cipherID + ".");
        } catch (MCRCryptCipherConfigurationException e) {
            LOGGER.error("Invalid configuration or key.", e);
        }
```
Perhaps an exception could also be caught and evaluated for the logs and set once.